### PR TITLE
fix: salesforce batch pagination

### PR DESCRIPTION
**Pull request by manas-vessel**
## Description

_Detailed description of the changes and the intent behind them_

## Checklist

- [ ] Changes are covered by tests if behavior has been changed or added
- [ ] Tests have 100% coverage
- [ ] If code changes were made, the version in `package.json` has been bumped (matching semver)

## Resolves

If the PR resolves an open issue tag it here. For example: `Resolves #34`

### CODE REVIEW

File: src/platforms/salesforce/actions/list-view-results/find.ts [hunk 1]
```diff
@@ -10,7 +11,7 @@ export default action(
     mutation: false,
     schema: z.object({
       id: z.string(),
-      objectType: z.string(),
+      objectType: salesforceSupportedObjectType,
     }),
```
Comment (averyyip) [inline src/platforms/salesforce/actions/list-view-results/find.ts:15]: Should the typing be tighter?

File: src/platforms/salesforce/actions/query-builder.ts [hunk 2]
```diff
@@ -102,15 +101,17 @@ export const salesforceQueryBuilder: Record<
       ${selectClauses.join(', ')}
       FROM ${objectType}
       WHERE Id IN ('${ids.join("','")}')
+      ORDER BY Id DESC
+      LIMIT ${limit}
```
Comment (averyyip) [inline src/platforms/salesforce/actions/query-builder.ts:105]: Should we only have this statement if the `limit` is provided as a param?

File: src/platforms/salesforce/actions/list-view-results/find.ts (hunk 1 above)
Comment (manas-vessel) [reply src/platforms/salesforce/actions/list-view-results/find.ts:15]: Tighter in the sense only allowing Leads and Contacts? They support listviews for all objects (except LitsView, of course)

Comment (averyyip) [reply src/platforms/salesforce/actions/list-view-results/find.ts:15]: ah ic perfect

**Review by averyyip — APPROVED** [review]

File: src/platforms/salesforce/actions/list-view-results/find.ts [hunk 3]
```diff
@@ -10,7 +11,7 @@ export default action(
     mutation: false,
     schema: z.object({
       id: z.string(),
-      objectType: z.string(),
+      objectType: salesforceSupportedObjectType,
```
Comment (zkirby) [inline src/platforms/salesforce/actions/list-view-results/find.ts:14]: my understanding was that we're removing cursor?

**Review by zkirby — CHANGES_REQUESTED** [review]

**Review by zkirby — APPROVED** [review]

**Pull request merged**